### DISCUSSION TITLE
Reset a model

### DISCUSSION THREAD

**Discussion by gintriago12** [online-ml/river#708] [Q&A] [answered]
Hi,

Is there any method to reset a model? For example, I am training a Hoeffding Adaptive Tree using batches of data, but for each batch I want the model to start learning from scratch. So, I am looking for something like this:

for batch, y in batches:
      model.reset()
      ypred = model.predict_proba_many(batch)
      metric.update(ypred,y)
      model.learn_many(batch,y)

Thanks,

**Comment by gbolmier** [comment, marked as answer]
Hi @gintriago12 :)

~~No there isn't.~~ What prevents you from defining a new model? For example, you could define a function to create a new model  and just call it whenever you want to *"reset"* it:

```python
def create_model():
    ...

model = create_model()
```

**Reply by MaxHalford** [reply]
Actually every River object (not just models) has a `.clone()` method :). I'll admit that's not well documented.

**Reply by gintriago12** [reply]
Thanks @gbolmier and @MaxHalford
